Sources/TuistKit/Commands/TestCommand.swift

```diff
@@ -61,15 +61,9 @@ struct TestCommand: AsyncParsableCommand {
     
     @Option(
         name: .long,
-        help: "Tests will run <number> of times."
+        help: "Tests will run <number> of times until they succeed."
     )
-    var testIterations: Int?
-    
-    @Flag(
-        name: .long,
-        help: "Tests will retry on failure. May be used in conjunction with -test-iterations <number>, in which case <number> will be the maximum number of iterations. Otherwise, a maximum of 3 is assumed."
-    )
-    var retryTestsOnFailure: Bool = false
+    var retryCount: Int = 0
```

Comment:
If `1` means trying just once, shouldn't this be 1 by default, and the parameter be added only if this is > 1?

Reply:
It's tricky because with a `retryCount` of 1 we're implying it would run 1 more time after a failure, which is probably why Apple went with `test-iterations`.

How about I change it to `testIterationsOnFailure` ?

Reply:
From the Apple documentation `-test-iterations <number>: If specified, tests will run <number> times. May be used in conjunction with either -retry-tests-on-failure or -run-tests-until-failure, in which case this will become the maximum number of iterations.`

I would expect that setting it to 2 It would run 2 times (the first one, and one retry). Could you please double check it?

Once we are sure about that, let's fix the default if needed. If it's as you said and `0` is ok, then we should fix the help, to say `Test will retry` instead of `Test will run`, and it might be worth to put an example there to remove any doubt

Reply:
That's correct, setting `-test-iterations 2` will run the tests twice (I've confirmed this in my own testing).

But since we're passing `-retry-tests-on-failure` it would only run once if the test passes the first time, but give it one more try if it failed the first time.

What if I pass `count+1` to the underlying `-test-iterations` argument to match our unique verbiage that's separate from Xcode?

Reply:
I think it's fine to pass count directly (I agree that with 2 we would try twice, and hence retry just once), but if we document well I think it's not an issue.
Maybe we could change the parameter name to `--max-retry-count` to make clear that if it pass it just executes once

Reply:
GTM, added a suggestion for the `help` to make it crystal clear